src/core/utils/matching/matchRequestUrl.ts

```diff
@@ -52,10 +52,13 @@ export function coercePath(path: string): string {
   )
 }
 
+const cache = new Map<string, Match>()
 /**
  * Returns the result of matching given request URL against a mask.
  */
 export function matchRequestUrl(url: URL, path: Path, baseUrl?: string): Match {
+  const key = `${url}|${path}|${baseUrl}`
```

Comment:
We also would want to use a better key than this, since we'll strip url params/hashes/etc before mathching.  Maybe we can extract some logic here and only do this 'url preparation' once per request vs per attempt to match